components/core/SlatePreviewBlockExternal.js

```diff
@@ -0,0 +1,436 @@
+          )}
+          <div css={STYLES_INFO}>
+            <div css={STYLES_OBJECT_COUNT}>{this.props.slate.data.objects.length}</div>
+            <div style={{ width: `85%` }}>
+              <div css={STYLES_TITLE_LINE}>
+                <div css={STYLES_TITLE}>{this.props.slate.data.name}</div>
+              </div>
+              {this.props.slate.data.body ? (
+                <div css={STYLES_BODY}>
+                  <ViewAllButton noButton fullText={this.props.slate.data.body} maxCharacter={100}>
+                    <ProcessedText text={this.props.slate.data.body} />
+                  </ViewAllButton>
+                </div>
+              ) : (
+                <div style={{ height: "8px" }} />
+              )}
+            </div>
+          </div>
+        </span>
+
+        <span css={STYLES_MOBILE_ONLY}>
+          <div
+            style={{
+              width: "100%",
+              height: `${this.state.windowWidth - 80}px`,
+            }}
+          >
+            {first ? (
+              <SlateMediaObjectPreview
+                blurhash={first.blurhash}
+                centeredImage
+                charCap={30}
+                type={first.type}
+                url={first.url}
+                title={first.title || first.name}
+              />
+            ) : (
+              <div
+                css={STYLES_PLACEHOLDER}
+                style={{
+                  backgroundImage: `url(${placeholder})`,
+                  ...this.props.imageStyle,
+                }}
+              />
+            )}
+          </div>
+          <div css={STYLES_INFO}>
+            <div css={STYLES_OBJECT_COUNT}>{this.props.slate.data.objects.length}</div>
+            <div style={{ width: `85%` }}>
+              <div css={STYLES_TITLE_LINE}>
+                <div css={STYLES_TITLE}>{this.props.slate.data.name}</div>
+              </div>
+            </div>
+          </div>
+        </span>
+      </div>
+    );
+  }
+}
+
+const STYLES_LINK = css`
+  color: ${Constants.system.black};
+  text-decoration: none;
+  width: calc(33.33% - 16px);
+  margin-bottom: 24px;
+
+  @media (max-width: ${Constants.sizes.tablet}px) {
+    width: 50%;
+  }
+
+  @media (max-width: ${Constants.sizes.mobile}px) {
+    width: 100%;
+  }
+`;
+
+const STYLES_SLATES = css`
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  overflow: hidden;
+  padding-bottom: 48px;
+  justify-content: space-between;
+
+  @media (max-width: ${Constants.sizes.mobile}px) {
+    display: block;
+  }
+`;
+
+export default class SlatePreviewBlocksExternal extends React.Component {
+  state = {
+    imageSize: 56,
+  };
+
+  componentDidMount = () => {
+    this.calculateWidth();
+    this.debounceInstance = this.debounce(this.calculateWidth, 350);
+    window.addEventListener("resize", this.debounceInstance);
+  };
+
+  componentWillUnmount = () => {
+    window.removeEventListener("resize", this.debounceInstance);
+  };
+
+  debounce = (fn, time) => {
```

Comment:
We should be using `~/common/window` `debounce so we don't reimplement debounce everywhere.

Reply:
for this one, do you mean addding `import * as Window from "~/common/window";`?